### PR TITLE
Refactor PI Network

### PR DESCRIPTION
**Pull request by michaelkad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
=== RUN   TestAccIBMPINetworkbasic
--- PASS: TestAccIBMPINetworkbasic (31.87s)
PASS

=== RUN   TestAccIBMPINetworkGatewaybasic
--- PASS: TestAccIBMPINetworkGatewaybasic (32.26s)
PASS

=== RUN   TestAccIBMPINetworkGatewaybasicSatellite
--- PASS: TestAccIBMPINetworkGatewaybasicSatellite (26.70s)
PASS

```

### CODE REVIEW

File: ibm/service/power/resource_ibm_pi_network.go [hunk 1]
```diff
@@ -44,94 +38,96 @@ func ResourceIBMPINetwork() *schema.Resource {
...
+				Computed:      true,
+				ConflictsWith: []string{Arg_NetworkJumbo},
+				Description:   "PI Maximum Transmission Unit",
+				Optional:      true,
+				Type:          schema.TypeInt,
+			},
+			Arg_NetworkName: {
+				Description: "PI network name",
+				Required:    true,
 				Type:        schema.TypeString,
```
Comment (ismirlia) [inline ibm/service/power/resource_ibm_pi_network.go:111]: ValidateFunc: validation.NoZeroValues, on required arguments?

Comment (michaelkad) [reply ibm/service/power/resource_ibm_pi_network.go]: Updated.

**Review by ismirlia — APPROVED** [review]

**Comment by ares123-cmyk** [comment]
hello ,can you me this is the communiticy about IBM and PI network , the can make connect about the blockchain on pi network??

**Comment by michaelkad** [comment]
> hello ,can you me this is the communiticy about IBM and PI network , the can make connect about the blockchain on pi network??

@ares123-cmyk Could you rephrase what you said?

**Review by faisalenz — APPROVED** [review]

File: ibm/service/power/resource_ibm_pi_network.go [hunk 2]
```diff
@@ -200,18 +199,25 @@ func resourceIBMPINetworkCreate(ctx context.Context, d *schema.ResourceData, met
...
 		body.IPAddressRanges = ipBodyRanges
 		body.Gateway = gateway
 		body.Cidr = networkcidr
 	}
+
+	if networktype == PubVlan {
+		if _, ok := d.GetOk(Arg_Cidr); ok {
+			return diag.Errorf("%s cannot be set when %s is dhcp-vlan or vlan", Arg_Cidr, Arg_NetworkType)
+		}
+	}
```
Comment (Alexander-Kita) [inline ibm/service/power/resource_ibm_pi_network.go:219]:
```suggestion
	if _, ok := d.GetOk(Arg_Cidr); ok && networktype == PubVlan {
		return diag.Errorf("%s cannot be set when %s is dhcp-vlan or vlan", Arg_Cidr, Arg_NetworkType)
	}
```

Just a suggestion to avoid unnecessary nested if

**Comment by ismirlia** [comment]
Opened: https://github.com/powervs-ibm/terraform-provider-ibm/pull/181 closing this one
